### PR TITLE
Extension Cleanup

### PR DESCRIPTION
**Pull request by LukasHiveMQ**
Cleaned up nullability annotations. Removed unused code. Updated JUnit from 4 to 5.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -187,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2018 dc-square GmbH
+   Copyright [yyyy] [name of copyright owner]
```
Comment (YannickWeber) [inline LICENSE:189]:
```suggestion
   Copyright 2018 HiveMQ
```

File: README.adoc [hunk 2]
```diff
@@ -5,65 +5,70 @@
 :maven-documentation-profile-link: http://maven.apache.org/guides/introduction/introduction-to-profiles.html
 :hivemq-support: http://www.hivemq.com/support/
 :hivemq-listener: https://www.hivemq.com/docs/hivemq/4.4/user-guide/listeners.html#tcp-listener
+:hivemq-extension-download: https://www.hivemq.com/extension/file-rbac-extension/
```
Comment (YannickWeber) [inline README.adoc:8]: you can have the GitHub link here

**Review by YannickWeber — APPROVED** [review]

File: README.adoc (hunk 2 above)
Comment (LukasHiveMQ) [reply README.adoc:8]: I aligned the links on all extensions to be the marketplace.

**Review by YannickWeber — APPROVED** [review]

File: README.adoc [hunk 3]
```diff
@@ -194,11 +198,15 @@ The amount of hashing iteration can be specified with the `-i` parameter.
 |===
 
 [#substitution]
-===== Substitution
-The special markers `${{clientid}}` and `${{username}}` in the topic filter for a permission are automatically replaced by the extension with the client identifier and username of the client for which authorization is performed. This allows to configure a permission that applies to multiple clients, but always contains their specific client identifier or username in the topic. Limiting each client to "their own" topics.
+=== Substitution
+
+// suppress inspection "AsciiDocAttributeShouldBeDefined"
```
Comment (SgtSilvio) [inline README.adoc:203]: this should probably not be suppressed, but properly escaped

File: README.adoc [hunk 4]
```diff
@@ -125,8 +130,7 @@ The credentials configuration includes the following settings.
 </file-rbac>
 ----
 
-===== User configuration
-
+=== User configuration
```
Comment (SgtSilvio) [inline README.adoc:133]:
```suggestion
=== User Configuration
```

File: README.adoc [hunk 5]
```diff
@@ -173,19 +176,20 @@ The amount of hashing iteration can be specified with the `-i` parameter.
 </user>
 ----
 
-===== Role configuration
+=== Role configuration
```
Comment (SgtSilvio) [inline README.adoc:179]:
```suggestion
=== Role Configuration
```

File: README.adoc [hunk 6]
```diff
@@ -194,11 +198,15 @@ The amount of hashing iteration can be specified with the `-i` parameter.
...
+=== Substitution
+
+// suppress inspection "AsciiDocAttributeShouldBeDefined"
+The special markers `${{clientid}}` and `${{username}}` in the topic filter for a permission are automatically replaced by the extension with the client identifier and username of the client for which authorization is performed.
+This allows to configure a permission that applies to multiple clients, but always contains their specific client identifier or username in the topic.
+Limiting each client to "their own" topics.
 
 [#extensions-config]
-==== Extension configuration
+=== Extension configuration
```
Comment (SgtSilvio) [inline README.adoc:209]:
```suggestion
=== Extension Configuration
```

File: README.adoc [hunk 7]
```diff
@@ -5,65 +5,70 @@
...
+The credentials configuration file is watched for changes and reloaded at runtime if necessary.
+If the credentials configuration file has changed and contains a valid configuration, then the previous configuration is automatically archived to an archive folder `credentials-archive` inside the extension folder.
+So that changes can be tracked and rolled-back if needed.
 If the new credentials configuration is invalid the current configuration is maintained.
 
 NOTE: The permissions for connected clients are not changed, only new connecting clients are affected.
 
 [#credentials-config]
-==== Credentials configuration
+=== Credentials configuration
```
Comment (SgtSilvio) [inline README.adoc:71]:
```suggestion
=== Credentials Configuration
```

File: README.adoc [hunk 8]
```diff
@@ -5,65 +5,70 @@
...
+:hivemq-extension-download: https://www.hivemq.com/extension/file-rbac-extension/
 
-== HiveMQ File Role based Access Control Extension
+= HiveMQ File Role based Access Control Extension
 
-*Extension Type*: Security
+image:https://img.shields.io/badge/Extension_Type-Security-orange?style=for-the-badge[Extension Type]
+image:https://img.shields.io/github/v/release/hivemq/hivemq-file-rbac-extension?style=for-the-badge[GitHub release (latest by date),link=https://github.com/hivemq/hivemq-file-rbac-extension/releases/latest]
+image:https://img.shields.io/github/license/hivemq/hivemq-file-rbac-extension?style=for-the-badge&color=brightgreen[GitHub,link=LICENSE]
+image:https://img.shields.io/github/workflow/status/hivemq/hivemq-file-rbac-extension/CI%20Check/master?style=for-the-badge[GitHub Workflow Status (branch),link=https://github.com/hivemq/hivemq-file-rbac-extension/actions/workflows/check.yml?query=branch%3Amaster]
```
Comment (SgtSilvio) [inline README.adoc:15]: badge can be fixed in this PR

File: src/main/java/com/hivemq/extensions/rbac/configuration/entities/ExtensionConfig.java [hunk 9]
```diff
@@ -36,14 +35,12 @@ public class ExtensionConfig {
...
-    @Nullable
     @XmlElementWrapper(name = "listener-names")
     @XmlElement(name = "listener-name")
-    private Set<String> listenerNames;
+    private @Nullable Set<String> listenerNames;
 
-    @NotNull
     @XmlElement(name = "password-type", defaultValue = "HASHED")
-    private PasswordType passwordType = PasswordType.HASHED;
+    private @Nullable PasswordType passwordType = PasswordType.HASHED;
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/rbac/configuration/entities/ExtensionConfig.java:43]: how can this get null?

File: src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java [hunk 10]
```diff
@@ -25,114 +25,82 @@
...
 
-    @Nullable
     @XmlElement(name = "topic", required = true)
-    private String topic;
+    private @Nullable String topic;
 
-    @NotNull
     @XmlElement(name = "activity", defaultValue = "ALL")
-    private TopicPermission.MqttActivity activity = TopicPermission.MqttActivity.ALL;
+    private @Nullable TopicPermission.MqttActivity activity = TopicPermission.MqttActivity.ALL;
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java:36]: how can this get null?

File: src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java [hunk 11]
```diff
@@ -25,114 +25,82 @@
...
 
-    @NotNull
     @XmlElement(name = "qos", defaultValue = "ALL")
-    private TopicPermission.Qos qos = TopicPermission.Qos.ALL;
+    private @Nullable TopicPermission.Qos qos = TopicPermission.Qos.ALL;
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java:39]: how can this get null?

File: src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java [hunk 12]
```diff
@@ -25,114 +25,82 @@
...
 
-    @NotNull
     @XmlElement(name = "retain", defaultValue = "ALL")
-    private TopicPermission.Retain retain = TopicPermission.Retain.ALL;
+    private @Nullable TopicPermission.Retain retain = TopicPermission.Retain.ALL;
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java:42]: how can this get null?

File: src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java [hunk 13]
```diff
@@ -25,114 +25,82 @@
...
 
-    @NotNull
     @XmlElement(name = "shared-subscription", defaultValue = "ALL")
-    private TopicPermission.SharedSubscription sharedSubscription = TopicPermission.SharedSubscription.ALL;
+    private @Nullable TopicPermission.SharedSubscription sharedSubscription = TopicPermission.SharedSubscription.ALL;
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java:45]: how can this get null?

File: src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java [hunk 14]
```diff
@@ -25,114 +25,82 @@
...
 
-    @NotNull
     @XmlElement(name = "shared-group", defaultValue = "#")
-    private String sharedGroup = "#";
+    private @Nullable String sharedGroup = "#";
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/extensions/rbac/configuration/entities/Permission.java:48]: how can this get null?

File: src/test/java/com/hivemq/extensions/rbac/utils/CredentialsValidatorTest.java [hunk 15]
```diff
@@ -113,118 +104,106 @@ public class CredentialsValidatorTest {
...
+    @BeforeEach
+    void setUp(@TempDir final @NotNull File extensionFolder) throws Exception {
+        this.extensionFolder = extensionFolder;
+        this.scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+        this.validator = initValidator(PLAIN_CREDENTIALS, false);
     }
 
-    @After
+    @AfterEach
     public void after() {
```
Comment (SgtSilvio) [inline src/test/java/com/hivemq/extensions/rbac/utils/CredentialsValidatorTest.java:119]:
```suggestion
    void after() {
```

File: README.adoc [hunk 16]
```diff
@@ -173,19 +176,20 @@ The amount of hashing iteration can be specified with the `-i` parameter.
...
+=== Permission configuration
 
 |===
 |Configuration |Default |Description
-|`topic` |-|The topic on which this permission should apply. Can contain standard MQTT wildcards `#` and `+`. Also special substitution with `${{clientid}}` and `${{username}}` is supported.
+// suppress inspection "AsciiDocAttributeShouldBeDefined"
+|`topic` |-|The topic on which this permission should apply. Can contain standard MQTT wildcards `#` and `+`. Also, special substitution with `${{clientid}}` and `${{username}}` is supported.
 |`activity` |`ALL` |The activity which this client can perform on this topic. Can be `PUBLISH`, `SUBSCRIBE` or `ALL`.
 |`qos` |`ALL` |The MQTT QoS which this client can publish/subscribe with on this topic. The value can be `ZERO`, `ONE`, `TWO`, `ZERO_ONE`, `ONE_TWO`, `ZERO_TWO` or `ALL`.
 |`retained` |`ALL` |If a message published on this topic can/must be retained. Values are `NOT_RETAINED`, `RETAINED` or `ALL`. This setting is only relevant for PUBLISH messages.
```
Comment (MicWalter) [inline README.adoc:195]: See PR #19 
```suggestion
|`retain` |`ALL` |If a message published on this topic can/must be retained. Values are `NOT_RETAINED`, `RETAINED` or `ALL`. This setting is only relevant for PUBLISH messages.
```

File: src/main/java/com/hivemq/extensions/rbac/configuration/entities/ExtensionConfig.java (hunk 9 above)
Comment (LukasHiveMQ) [reply src/main/java/com/hivemq/extensions/rbac/configuration/entities/ExtensionConfig.java:43]: It is possible to generate cases with JAXB where writing a wrong value will result in null.

**Pull request merged**